formula-android/src/main/java/com/instacart/formula/android/internal/FragmentLifecycle.kt

```diff
@@ -37,8 +39,28 @@ private fun Fragment.getFragmentKey(): FragmentKey {
     return fragment?.getFragmentKey() ?: EmptyFragmentKey(tag.orEmpty())
 }
 
+/**
+ * Gets a persisted across configuration changes fragment identifier or initializes
+ * one if it doesn't exist.
+ */
 private fun Fragment.getFragmentInstanceId(): String {
-    return arguments?.getString(FormulaFragment.ARG_FORMULA_ID) ?: ""
+    return if (this is BaseFormulaFragment<*>) {
+        val arguments = arguments ?: run {
+            Bundle().apply {
+                arguments = this
+            }
+        }
+        val id = arguments.getString(FormulaFragment.ARG_FORMULA_ID, "")
+        if (id.isNullOrBlank()) {
+            val initializedId = UUID.randomUUID().toString()
+            arguments.putString(FormulaFragment.ARG_FORMULA_ID, initializedId)
+            initializedId
+        } else {
+            id
+        }
+    } else {
+        ""
+    }
```

Comment:
Should we cover this logic with tests?

Reply:
That's a good question - we already cover this logic via tests in the `formula-android` and `formula-android-tests` modules. Specifically, the most important tests are the following:
- https://github.com/instacart/formula/blob/877827fff30fdf3065fd7ee9c822f57ec6ce57f2/formula-android-tests/src/test/java/com/instacart/formula/FragmentFlowRenderViewTest.kt#L146
- https://github.com/instacart/formula/blob/877827fff30fdf3065fd7ee9c822f57ec6ce57f2/formula-android-tests/src/test/java/com/instacart/formula/FragmentFlowRenderViewTest.kt#L162